packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -63,9 +107,11 @@ def test_ip_per_container(cluster):
         ensure_routable(cmd, service_points)()
 
 
+@skip_if_lb_disabled
 def test_ip_per_container_with_named_vip(cluster):
     """Test if we are able to connect to a task with ip-per-container mode using named vip
     """
+    test_if_overlay_ok(cluster)
```

Comment:
It looks like you are trying to use this as a setup fixture. The proper form would be:

```
@pytest.fixture
def overlay_ok():
    # do all of your stuff
```
then, in your test:
```
@skip_if_lb_disabled
def test_ip_per_container_with_named_vip(cluster, overlay_ok):
    # do you test code
```
By naming the fixture as your argument in the test function, pytest will ensure that the code is run before this test. Furthermore, when code is to be repeated (and not expected to change) you can do this with pytest.fixture(scope='session') to only run it once
I think this is what you are intending and is the proper way to use setup fixture